### PR TITLE
Task/38 pre built mage for common deploy workflow

### PR DESCRIPTION
**Pull request by 12345pradeep**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Calance generic CI-CD
+on:
+  push:
+    tags:
+    - v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+
+    branches:
```
Comment (agoyalcalance) [inline .github/workflows/build.yml]: remove the build trigger for branches... let us keep it consistent with tags.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - 'main'
+    paths:
+    - 'Dockerfile'
+    - '**.py'
+  workflow_dispatch:
+
+
+jobs:
+  build-docker:
```
Comment (agoyalcalance) [inline .github/workflows/build.yml]: can't you call the `docker build` job from `calance-workflows` common workflows repository?

File: action.yml [hunk 3]
```diff
@@ -22,13 +22,13 @@ outputs:
     description: 'Build status'
 runs:
   using: 'docker'
-  image: 'Dockerfile'
+  image: 'docker://ghcr.io/calance-us/build-jenkins-job-image:v0.0.0-rc0'
```
Comment (agoyalcalance) [inline action.yml]: this version should not be v0.0.0-rc0 it should be a production version. 1.0.0 or something like that.

**Review by agoyalcalance — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (12345pradeep) [reply .github/workflows/build.yml]: done

File: .github/workflows/build.yml (hunk 2 above)
Comment (12345pradeep) [reply .github/workflows/build.yml]: yes we can, i have changed it too

File: action.yml (hunk 3 above)
Comment (12345pradeep) [reply action.yml]: done

**Review by ypmittal — APPROVED** [review]
LGTM

**Pull request merged**